libsel4vm/src/arch/arm/vgic/vgic.c

```diff
@@ -1156,13 +1127,11 @@ int vm_install_vgic(vm_t *vm)
 
 int vm_vgic_maintenance_handler(vm_vcpu_t *vcpu)
 {
-    int idx;
-    int err;
-    idx = seL4_GetMR(seL4_UnknownSyscall_ARG0);
+    int idx = seL4_GetMR(seL4_UnknownSyscall_ARG0);
     /* Currently not handling spurious IRQs */
     assert(idx >= 0);
 
-    err = handle_vgic_maintenance(vcpu, idx);
+    int err = handle_vgic_maintenance(vcpu, idx);
```

Comment:
Maybe we should also log a message here if this fails, as things are badly broken then. See also comment https://github.com/seL4/seL4_projects_libs/pull/28/files#r824617661

Reply:
I think we should have a way to halt the VM execution at this point -- I think it should execute correctly or not execute at all, something in between only results in frustation and wasted time debugging something flaky.

Reply:
It's a lost interrupt in this case, which is not necessarily this fatal, but very annoying. Logging is the best one can do to allow post-mortem analysis. The error behavior should become a follow-up discussion to leave out od this PR, otherwise this never gets merged ;)

Reply:
True -- actually that reminds me I have pondered whether these readability improvements should be a separate PR (or moving the real bugfix into its own PR).

Reply:
From my side, having this in a separate commit is sufficient. It's more about how to get the others to agree to merge all the improvements, there are a few other PRs now also that seem useful (https://github.com/seL4/seL4_projects_libs/pull/29, https://github.com/seL4/seL4_projects_libs/pull/31 ...)
Opinions, @yyshen, @kent-mcleod, @lsf37